src/modules/stakewise/stakewise-daemon/server/status/get-validator-statuses.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/nodeset-org/hyperdrive/daemon-utils/server"
+)
+
+// ===============
+// === Factory ===
+// ===============
+
+type statusGetValidatorsStatusesContextFactory struct {
+	handler *StatusHandler
+}
+
+func (f *statusGetValidatorsStatusesContextFactory) Create(args url.Values) (*statusGetValidatorsStatusesContext, error) {
+	c := &statusGetValidatorsStatusesContext{
+		handler: f.handler,
+	}
+	inputErrs := []error{}
+	return c, errors.Join(inputErrs...)
+}
+
+func (f *statusGetValidatorsStatusesContextFactory) RegisterRoute(router *mux.Router) {
+	server.RegisterQuerylessGet[*statusGetValidatorsStatusesContext, swapi.ValidatorStatusData](
+		router, "status", f, f.handler.serviceProvider.ServiceProvider,
+	)
+}
+
+// ===============
+// === Context ===
+// ===============
+
+type statusGetValidatorsStatusesContext struct {
+	handler *StatusHandler
+}
+
+func (c *statusGetValidatorsStatusesContext) PrepareData(data *swapi.ValidatorStatusData, opts *bind.TransactOpts) error {
+	sp := c.handler.serviceProvider
+	bc := sp.GetBeaconClient()
+	w := sp.GetWallet()
+	nc := sp.GetNodesetClient()
+	registeredPubkeys, err := nc.GetRegisteredValidators()
+	if err != nil {
+		return fmt.Errorf("error getting registered validators: %w", err)
+	}
+	privateKeys, err := w.GetAllPrivateKeys()
+	if err != nil {
+		return fmt.Errorf("error getting private keys: %w", err)
+	}
+	publicKeys, err := w.DerivePubKeys(privateKeys)
+	if err != nil {
+		return fmt.Errorf("error getting public keys: %w", err)
+	}
+	statuses, err := bc.GetValidatorStatuses(context.Background(), publicKeys, nil)
+	if err != nil {
+		return fmt.Errorf("error getting validator statuses: %w", err)
+	}
+	var generatedValidators, uploadedNodesetValidators, uploadedStakewiseValidators, registeredStakewiseValidators, withdrawalDone, withdrawalPossible, exitedSlashed, exitedUnslashed, activeSlashed, activeExited, activeOngoing, pendingQueued, pendingInitialized []beacon.ValidatorPubkey
+
+	for _, pubKey := range publicKeys {
+		if IsWithdrawalDone(pubKey, statuses) {
+			withdrawalDone = append(withdrawalDone, pubKey)
+		} else if IsWithdrawalPossible(pubKey, statuses) {
+			withdrawalPossible = append(withdrawalPossible, pubKey)
+		} else if IsExitedSlashed(pubKey, statuses) {
+			exitedSlashed = append(exitedSlashed, pubKey)
+		} else if IsExitedUnslashed(pubKey, statuses) {
+			exitedUnslashed = append(exitedUnslashed, pubKey)
+		} else if IsActiveSlashed(pubKey, statuses) {
+			activeSlashed = append(activeSlashed, pubKey)
+		} else if IsActiveExited(pubKey, statuses) {
+			activeExited = append(activeExited, pubKey)
+		} else if IsActiveOngoing(pubKey, statuses) {
+			activeOngoing = append(activeOngoing, pubKey)
+		} else if IsPendingQueued(pubKey, statuses) {
+			pendingQueued = append(pendingQueued, pubKey)
+		} else if IsPendingInitialized(pubKey, statuses) {
+			pendingInitialized = append(pendingInitialized, pubKey)
+		} else if IsRegisteredToStakewise(pubKey, statuses) {
```

Comment:
You probably want to make these `if` instead of `else if`; like the Stakewise / Nodeset validator status should be separate from the Beacon status.

Reply:
I see. I was thinking that there would be a single state and a pubkey goes through these states in a linear fashion. I can update this to have two separate states